src/sentry/incidents/logic.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import
+
+from django.db import transaction
+
+from sentry.api.event_search import get_snuba_query_args
+from sentry.incidents.models import (
+    Incident,
+    IncidentGroup,
+    IncidentProject,
+)
+from sentry.utils.snuba import (
+    raw_query,
+    SnubaTSResult,
+)
+
+
+def create_incident(
+    organization,
+    status,
+    title,
+    query,
+    date_started,
+    date_detected=None,
+    detection_uuid=None,
+    projects=None,
+    groups=None,
+):
+    if date_detected is None:
+        date_detected = date_started
+
+    with transaction.atomic():
+        incident = Incident.objects.create(
+            organization=organization,
+            detection_uuid=detection_uuid,
+            status=status.value,
+            title=title,
+            query=query,
+            date_started=date_started,
+            date_detected=date_detected,
+        )
+        if projects:
+            IncidentProject.objects.bulk_create([
+                IncidentProject(incident=incident, project=project) for project in projects
+            ])
+        if groups:
+            IncidentGroup.objects.bulk_create([
+                IncidentGroup(incident=incident, group=group) for group in groups
+            ])
+    return incident
+
+
+def build_incident_query_params(incident):
+    params = {'start': incident.date_started, 'end': incident.current_end_date}
+    group_ids = list(set([group.id for group in incident.groups.all()]))
```

Comment:
Do you need to hydrate all the groups if you're just picking out their ids?

Reply:
Fair, I'll fix this